### PR TITLE
Add tests for annotation delete popup

### PR DESCRIPTION
**Pull request by Aditi-1400**
Left some inline comments, also open for better test-naming suggestions.
To Do:
- [x] Change `xit` to `it`, `fdescribe` to `describe`
- [x] Improve tests for pop-up message.
- [x] Add tests for stamp editor:
    - [x] when deleting an image
    - [x] when deleting both an highlight and an image at the same time 
 - [x]  Lint Fixes

### CODE REVIEW

File: test/integration/highlight_editor_spec.mjs [hunk 1]
```diff
@@ -2144,4 +2145,202 @@ describe("Highlight Editor", () => {
       );
     });
   });
+
+  fdescribe("Annotation pop-up has the expected behaviour", () => {
```
Comment (Aditi-1400) [inline test/integration/highlight_editor_spec.mjs]: This needs to be changed to describe, keeping it fdescribe for now, so it's easier to review!

File: test/integration/highlight_editor_spec.mjs [hunk 2]
```diff
@@ -2144,4 +2145,202 @@ describe("Highlight Editor", () => {
...
+          await page.waitForSelector("#editorUndoBar:not([hidden])");
+
+          await page.waitForSelector("#editorUndoBarCloseButton");
+          await page.click("#editorUndoBarCloseButton");
+          await page.waitForSelector("#editorUndoBar", { hidden: true });
+        })
+      );
+    });
+
+    xit("must check that the popup disappears when a new annotation is created", async () => {
```
Comment (Aditi-1400) [inline test/integration/highlight_editor_spec.mjs]: This test is failing

File: test/integration/highlight_editor_spec.mjs [hunk 3]
```diff
@@ -2144,4 +2145,202 @@ describe("Highlight Editor", () => {
...
+          await waitForSerialized(page, 0);
+          await page.waitForSelector("#editorUndoBar:not([hidden])");
+
+          await kbSave(page);
+          await page.waitForSelector("#editorUndoBar", { hidden: true });
+        })
+      );
+    });
+
+    xit("must check that the popup disappears when an option from the 'more' menu is selected", async () => {
```
Comment (Aditi-1400) [inline test/integration/highlight_editor_spec.mjs]: This test is failing

File: test/integration/highlight_editor_spec.mjs [hunk 4]
```diff
@@ -2144,4 +2145,202 @@ describe("Highlight Editor", () => {
...
+          await page.waitForSelector("#editorUndoBar:not([hidden])");
+
+          await page.click('button[title="Tools"]');
+          await page.click("#lastPage");
+          await page.waitForSelector("#editorUndoBar", { hidden: true });
+        })
+      );
+    });
+
+    xit("must display the correct message for highlights", async () => {
```
Comment (Aditi-1400) [inline test/integration/highlight_editor_spec.mjs]: This test needs more work:
- Need to change it for localized text
- Also, the text is `Annotation removed` the first time, so I need to find a way to accurately check for this

File: test/integration/highlight_editor_spec.mjs [hunk 5]
```diff
@@ -2144,4 +2145,200 @@ describe("Highlight Editor", () => {
...
+          await waitForSerialized(page, 1);
+          await page.waitForSelector(getEditorSelector(0));
+          await page.waitForSelector(
+            `.page[data-page-number = "1"] svg.highlight[fill = "#FFFF00"]`
+          );
+        })
+      );
+    });
+
+    it("must check that the popup disappears when the close button is clicked", async () => {
```
Comment (nicolo-ribaudo) [inline test/integration/highlight_editor_spec.mjs:2219]: We should probably also test that the popup disappears when clicking on Undo

File: test/integration/highlight_editor_spec.mjs (hunk 4 above)
Comment (nicolo-ribaudo) [reply test/integration/highlight_editor_spec.mjs]: I _think_ all the tests run in english, so it shouldn't be affected by different locales.

**Review by nicolo-ribaudo — COMMENTED** [review]
Tests look good!

File: test/integration/highlight_editor_spec.mjs [hunk 6]
```diff
@@ -2144,4 +2145,272 @@ describe("Highlight Editor", () => {
...
+            { timeout: 1000 }
+          );
+
+          const message = await page.waitForSelector("#editorUndoBarMessage");
+          const messageText = await page.evaluate(
+            el => el.textContent,
+            message
+          );
+
+          expect(messageText).toContain(`\u200B2 annotations removed\u200B`);
```
Comment (Aditi-1400) [inline test/integration/highlight_editor_spec.mjs]: So, this text is not passing, I've tried logging messageText with character codes, turns out there's zero-width spaces on both sides of "2 annotations removed", I've tried a bunch of things to remove them, this still doesn't work!  :(
Suggestions are welcome

**Review by nicolo-ribaudo — APPROVED** [review]
Looks good :) This can be squash-merged.

**Pull request merged**